### PR TITLE
add .gitignore

### PR DESCRIPTION
**Pull request by rleungx**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
+.idea/
+*.iml
+out/
+
+# Vscode files
+.vscode/
+
+target
```
Comment (sid0) [inline .gitignore]: I'm going to add a trailing slash to this.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# OSX trash
+.DS_Store
```
Comment (sid0) [inline .gitignore]: I'll add a trailing slash here.

Comment (rleungx) [reply .gitignore]: According to [this](https://github.com/github/gitignore/blob/master/Global/macOS.gitignore), perheps .DS_Store is enough.

Comment (sid0) [reply .gitignore]: Ah indeed, looks like `.DS_Store` is a file and not a directory.

**Pull request closed without merging**